### PR TITLE
Split input context & variable utilities

### PR DESCRIPTION
**Pull request by guusw**
Some preliminary changes for the world space UI.
Here I made references to internal context variables composable using the RequiredContextVariable template instead of the BaseConsumer base class.
Also split the input context from the graphics context so they can be provided elsewhere.

### CODE REVIEW

**Review by sinkingsugar — APPROVED** [review]
Ok if CI is clean

File: src/core/exposed_type_utils.hpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+//   SHExposedTypesInfo requiredVariables() {
+//     static auto e = exposedTypesOf(SomeExposedTypeInfo, SHExposedTypeInfo{...});
+//     return e;
+//   }
+//
+//   RequiredContextVariable<GraphicsContext, GraphicsContext::Type, Base::graphicsContextVarName> graphicsContext;
+//   SHExposedTypesInfo requiredVariables() {
+//     static auto e = exposedTypesOf(decltype(graphicsContext)::getExposedTypeInfo());
+//     return e;
+//   }
```
Comment (Kryptos-FR) [inline src/core/exposed_type_utils.hpp:77]: Remove commented code.

**Review by Kryptos-FR — APPROVED** [review]
LGTM, a few cleanup suggestions.

File: src/core/exposed_type_utils.hpp (hunk 1 above)
Comment (guusw) [reply src/core/exposed_type_utils.hpp:77]: It's supposed to be an example, although I agree it looks like commented out code 😄

Comment (Kryptos-FR) [reply src/core/exposed_type_utils.hpp:77]: My bad.

File: src/extra/inputs.cpp [hunk 2]
```diff
@@ -3,16 +3,36 @@
...
+  void baseWarmup(SHContext *context) { _inputContext.warmup(context); }
+  void baseCleanup() { _inputContext.cleanup(); }
+  SHExposedTypesInfo baseRequiredVariables() {
+    static auto e = exposedTypesOf(gfx::RequiredGraphicsContext::getExposedTypeInfo());
+    return e;
+  }
+
+  SHExposedTypesInfo requiredVariables() { return baseRequiredVariables(); }
+  void warmup(SHContext *context) { baseWarmup(context); }
+  void cleanup() { baseCleanup(); }
```
[the remark this answers is not in the document]
Comment (guusw) [reply src/extra/inputs.cpp:33]: Removed those that didn't do anything besides the base functionality

**Pull request merged**